src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -40,6 +41,11 @@ public class SecurityTokenDescriptor
         /// </summary>
         public string Audience { get; set; }
 
+        /// <summary>
```

Comment:
We should take into account the properties such as Audience when creating the payload.
Give priority to SecurityTokenDescriptor is a claim is found of the same type.
May want to log 'information' that a duplicate exists.

Continue accounting for 'SetExpirationTimeOnTokenCreation' flag.

Reply:
Done. I also added in tests to make sure that the properties on SecurityTokenDescriptor are being used as expected.